src/main/python/ayab/pattern_import.py

```diff
@@ -540,15 +549,16 @@ def pattern2im(self, filename, palfilename=None):
             color_maxblue = getWordAt(color_data, 18)
             block = 0
             offset = 0
-            for c in range(color_maxindex):
+            for cs in range(color_maxindex):
                 if (offset + 3 > 512):
                     offset = 0
                     block += 512
                 index = color_start + block + offset
                 r = getByteAt(color_data, index)
                 g = getByteAt(color_data, index + 1)
                 b = getByteAt(color_data, index + 2)
-                self.colors.add(Color(n=c, r=r, g=g, b=b))
+                
+                self.colors[cs] = (Color(n=np.uint8(cs), r=r, g=g, b=b))
```

Comment:
This one _feels_ like it was a bug, but I could've misunderstood. `dict` doesn't **have** `.add` so this wouldn't've worked? It's an import function for a file I have no examples of, so I can't check if this actually fixes anything.